components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodActiveLanguageHook.kt

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package io.gitpod.jetbrains.remote
+
+import com.intellij.codeWithMe.ClientId
+import com.intellij.openapi.components.service
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManagerListener
+import com.intellij.openapi.fileTypes.LanguageFileType
+import com.intellij.openapi.vfs.VirtualFile
+import io.gitpod.jetbrains.remote.services.EventTrackerService
+
+class GitpodActiveLanguageHook : FileDocumentManagerListener {
```

Comment:
Does it guarantee to fire only when a file really has an editor and editor has a focus?

I wonder whether https://upsource.jetbrains.com/idea-ce/file/idea-ce-4d741bc560dd19306d4624d7c8a88aea537f4e6f/platform/analysis-api/src/com/intellij/openapi/fileEditor/FileEditorManagerListener.java is better? Did you try it? It seems to be project level service meaning that it at least need a window opened

Reply:
I think FileEditorManagerListener should e a proper one, highlighting is based on it: https://github.com/JetBrains/intellij-community/blob/1092934af23e578bd694dfe9d550316795582e30/platform/lang-impl/src/com/intellij/analysis/problemsView/toolWindow/HighlightingPanel.kt#L25